docs: better `global ignores` instruction (#19297)
eslint/eslint: docs/src/use/configure/configuration-files.md

* `name` - A name for the configuration object. This is used in error messages and config inspector to help identify which configuration object is being used. ([Naming Convention](#configuration-naming-conventions))
* `files` - An array of glob patterns indicating the files that the configuration object should apply to. If not specified, the configuration object applies to all files matched by any other configuration object.

> * `ignores` - An array of glob patterns indicating the files that the configuration object should not apply to. If not specified, the configuration object applies to all files matched by `files`. [global ignores](#globally-ignoring-files-with-ignores) is what you need if you want to apply the same `ignores` rule to every configuration objects.

Reviewer: ```suggestion
* `ignores` - An array of glob patterns indicating the files that the configuration object should not apply to. If not specified, the configuration object applies to all files matched by `files`. If `ignores` is used without any other keys in the configuration object, then the patterns act as [global ignores](#globally-ignoring-files-with-ignores) and it gets apply to every configuration object.
```

I think this much modification is good?

> * `ignores` - An array of glob patterns indicating the files that the configuration object should not apply to. If not specified, the configuration object applies to all files matched by `files`. If `ignores` is used without any other keys in the configuration object, then the patterns act as [global ignores](#globally-ignoring-files-with-ignores) and it gets apply to every configuration object.

Reviewer 2: ```suggestion
* `ignores` - An array of glob patterns indicating the files that the configuration object should not apply to. If not specified, the configuration object applies to all files matched by `files`. If `ignores` is used without any other keys in the configuration object, then the patterns act as [global ignores](#globally-ignoring-files-with-ignores) and it gets applied to every configuration object.
```

#### Globally ignoring files with `ignores`

Depending on how the `ignores` property is used, it can behave as non-global `ignores` or as global `ignores`.

> * If `ignores` is used without any other keys in the configuration object, then the patterns act as global ignores. This way it gets apply to every configuration object (not only to the configration object in which is defined). Global `ignores` allows you not to have to copy and keep the `ignores` property synchronized in more than one configuration object.

Reviewer 2: ```suggestion
* When `ignores` is used without any other keys (besides `name`) in the configuration object, then the patterns act as global ignores. This means they apply to every configuration object (not only to the configuration object in which it is defined). Global `ignores` allows you not to have to copy and keep the `ignores` property synchronized in more than one configuration object.
```

> * If `ignores` is used in conjunction with other properties in the same configuration object, then the patterns act as non-global ignores. This way `ignores` applies only that configuration object

Reviewer 2: ```suggestion
* If `ignores` is used with other properties in the same configuration object, then the patterns act as non-global ignores. This way `ignores` applies only to the configuration object in which it is defined.
```

Global and non-global `ignores` have some usage differences:

* patterns in non-global `ignores` only match the files (`dir/filename.js`) or files within directories (`dir/**`)

> * patterns in global `ignores` can also match directories (`dir/`), other than files like "non-global".

Reviewer 2: ```suggestion
* patterns in global `ignores` can match directories (`dir/`) in addition to the patterns that non-global ignores supports.
```

> In any case (non-global or global):

Reviewer 2: ```suggestion
For all uses of `ignores`:
```

> * the glob patterns you define are added after the default patterns, which are `["**/node_modules/", ".git/"]`

Reviewer 2: ```suggestion
* The patterns you define are added after the default ESLint patterns, which are `["**/node_modules/", ".git/"]`.
```

> * glob patterns always match files and directories that begin with a dot, such as `.foo.js` or `.fixtures`, unless those files are explicitly ignored. The only dot directory ignored by default is `.git`

Reviewer 2: ```suggestion
* The patterns always match files and directories that begin with a dot, such as `.foo.js` or `.fixtures`, unless those files are explicitly ignored. The only dot directory ignored by default is `.git`.
```

#### Globally ignoring files with `ignores`

> If you want to globally ignores some directory/files, and apply this directive to all configuration objects of the array,  `global ignores` is exactly made for this scenario.
> If `ignores` is used without any other keys in the configuration object, then the patterns act as global ignores. Here's an example:

Reviewer: ```suggestion
If `ignores` is used without any other keys in the configuration object, then the patterns act as global ignores. This way it gets apply to every configuration object. While patterns in non-global `ignores` only match the files or files within directories, patterns in global `ignores` can also match directories. Here's an example:
```

Let's try to match this with above description a bit.

Author: I made a big-rewrite of this section in https://github.com/eslint/eslint/pull/19297/commits/c6efc5197988afd344f6e54d788c8a4c7c4d1f48, check that

#### Globally ignoring files with `ignores`

Depending on how the `ignores` property is used, it can behave as non-global `ignores` or as global `ignores`.

> * When `ignores` is used without any other keys (besides `name`) in the configuration object, then the patterns act as global ignores. This means they apply to every configuration object (not only to the configuration object in which it is defined). Global `ignores` allows you not to have to copy and keep the `ignores` property synchronized in more than one configuration object.

Reviewer 3: I think the "global ignores apply to every configuration object" point of view is confusing as it implies a relation between global and non-global ignores, e.g., that non-global ignores can unignore files ignored by global ignores or vice versa, which isn't the case. I think we should keep global and non-global ignores described as two separate concepts:

1. Global ignores specify which files should be entirely ignored.
2. Non-global ignores specify which files the config object should not apply to.

Author: I'm not sure if I understood this correctly, and I don't know the internals.

Let's create an example to tackle this...

```js
// eslint.config.js

// Mix global and non-global ignores
export default [
    {
      name: "global-ignores",
      ignores: [".config/", "dist/", "tsconfig.json"]
      // This globally ignores ".config/", "dist/", "tsconfig.json"
    },
    {
      name: "non-global-ignores-1",
      ignores: [".eslint.config.js"],
      rules: {... }
      // This inherits globally ignores of ".config/", "dist/", "tsconfig.json"
      // and also ignore (only for this config object) ".eslint.config.js"
    },
];
```

Is this correct ?

> that non-global ignores can unignore files ignored by global ignores

Can you create a code example (like the one above) that does that ?

Reviewer 3: > > that non-global ignores can unignore files ignored by global ignores
> 
> Can you create a code example (like the one above) that does that ?

It doesn't work, but it could be understood that a config like this would enable linting of `tsconfig.json`:

```js
// eslint.config.js

// Mix global and non-global ignores
export default [
    {
      name: "global-ignores",
      ignores: [".config/", "dist/", "tsconfig.json"]
      // This globally ignores ".config/", "dist/", "tsconfig.json"
    },
    {
      name: "non-global-ignores-1",
      ignores: ["!tsconfig.json"], // attempts to apply this config to tsconfig.json and enable its linting
      rules: {... }
    },
];
```

Author: Clear!
 to discourage this pattern I would add a text in the section that explain the usage of "!pattern" if it is present in the docs.

If not present, this should be close to one of the first text that it's about "glob pattern" usage 

This can be a new PR
